src/benchmark_runner.cc

```diff
@@ -401,6 +403,44 @@ void BenchmarkRunner::RunWarmUp() {
   }
 }
 
+MemoryManager::Result* BenchmarkRunner::RunMemoryManager(
+    IterationCount& memory_iterations) {
+  // TODO(vyng): Consider making BenchmarkReporter::Run::memory_result an
+  // optional so we don't have to own the Result here.
+  // Can't do it now due to cxx03.
+  memory_results.push_back(MemoryManager::Result());
+  MemoryManager::Result* memory_result = &memory_results.back();
+  // Only run a few iterations to reduce the impact of one-time
+  // allocations in benchmarks that are not properly managed.
+  memory_iterations = std::min<IterationCount>(16, iters);
+  memory_manager->Start();
+  std::unique_ptr<internal::ThreadManager> manager;
+  manager.reset(new internal::ThreadManager(1));
+  b.Setup();
+  RunInThread(&b, memory_iterations, 0, manager.get(),
+              perf_counters_measurement_ptr);
+  manager->WaitForAllThreads();
+  manager.reset();
+  b.Teardown();
+  memory_manager->Stop(*memory_result);
+  return memory_result;
+}
+
+void BenchmarkRunner::RunProfilerManager() {
+  // TODO: Provide a way to specify the number of iterations.
+  IterationCount profile_iterations = 1;
+  std::unique_ptr<internal::ThreadManager> manager;
+  manager.reset(new internal::ThreadManager(1));
+  b.Setup();
+  profiler_manager->AfterSetupStart();
```

Comment:
why doesn't this report alongside the other results?

Reply:
Interesting question. What results make sense to report here?

Reply:
i'm not sure what the trace manager would return.  what do you expect to be reported?  for memory manager, no matter what the implementation, we expect number of allocations, max bytes used, total allocated, etc to be common to all memory management.

is there anything the profile manager might have that's common?

Reply:
Tracing will generate some output files, so it will want to report the names of the generated files.
The question is how to provide a generic way to communicate random bits of info back to the framework (if the framework wants to be the one to report the results to the user).
In the absence of a way to report back results, I was thinking trace manager would just use printf.

`CreateRunReport` takes `memory_iterations` and `memory_results` as arguments.
We could add `profile_iterations` and `profile_results`, we just need to figure out the type of `profile_results`.
There is `UserCounters` in `State`, so `ProfilerManager` can use that (I think, right?).
It doesn't support string text though. Maybe we could add a simple API that ProfilerManager can use to add some string text to the report? Eg, change `Stop` to return a string to report back to the user?

Reply:
if the tracing creates the output directly then there's no need to report them here as well.  i don't know if that's generally true, but we can certainly start with that as a baseline.

Reply:
SGTM